fn/slice.go

```diff
@@ -65,6 +65,32 @@ func Filter[A any](pred Pred[A], s []A) []A {
 	return res
 }
 
+// FilterMap takes a function argument that optionally produces a value and
+// returns a slice of the 'Some' return values.
+func FilterMap[A, B any](as []A, f func(A) Option[B]) []B {
+	bs := make([]B, 0)
+	for _, a := range as {
+		f(a).WhenSome(func(b B) {
+			bs = append(bs, b)
+		})
+	}
+	return bs
```

Comment:
surprised the linter isn't complaining about newline before return - did we turn that off 🤔